src/Hint/Duplicate.hs

```diff
@@ -70,9 +89,13 @@ add pos (v:vs) (Dupe p mp) = Dupe p $ Map.insertWith f v (add pos vs $ Dupe pos
 duplicateOrdered :: (Ord pos, Default pos, Ord val) => Int -> [[(pos,val)]] -> [(pos,pos,[val])]
```

Comment:
If you do `forall pos val .` on this line, then you can avoid restating the `Ord`/`Default` constraints on the lines below. You might need to turn on `ScopedTypeVariables`.

Reply:
Ok. Thanks for the hint.

Reply:
Done.